src/frozendict/core.py

```diff
@@ -50,11 +52,12 @@ def __new__(e4b37cdf_d78a_4632_bade_6f0579d8efac, *args, **kwargs):
             # empty singleton - start
             
             if self.__class__ == frozendict and not len(self):
-                try:
-                    self = cls.empty
+                global _empty_frozendict_singleton
```

Comment:
Can't `global` be avoided?

Reply:
Unfortunately not, without the global you get `UnboundLocalError: cannot access local variable '_empty_frozendict' where it is not associated with a value` due to the assignment at a later point in the same scope.  Python is weird.